internal/config/merge_test.go

```diff
@@ -0,0 +1,283 @@
+// Copyright 2024 Adevinta
+
+package config
+
+import (
+	"testing"
+
+	agentconfig "github.com/adevinta/vulcan-agent/config"
+	"github.com/google/go-cmp/cmp"
+)
+
+func TestLavaMerger_Merge(t *testing.T) {
+	tests := []struct {
+		name    string
+		cfg1    Config
+		cfg2    Config
```

Comment:
Rename to `dst` and `src` to make the test easier to follow.

```suggestion
		dst    Config
		src    Config
```